rust/sbp/src/sbp_tools.rs

```diff
@@ -0,0 +1,259 @@
+use swiftnav_rs::time::GpsTime;
+
+use crate::{
+    messages::{SBPMessage, SBP},
+    time::{MessageTime, RoverTime},
+    Result,
+};
+
+pub trait SBPTools: Iterator {
+    fn ignore_errors(self) -> HandleErrorsIter<Self, fn(&crate::Error) -> bool>
+    where
+        Self: Iterator<Item = Result<SBP>> + Sized,
+    {
+        HandleErrorsIter::new(self, |_| true)
+    }
+
+    fn log_errors(self) -> HandleErrorsIter<Self, fn(&crate::Error) -> bool>
+    where
+        Self: Iterator<Item = Result<SBP>> + Sized,
+    {
+        HandleErrorsIter::new(self, |e| {
+            log::warn!("{}", e);
+            true
+        })
+    }
+
+    fn handle_errors<F>(self, on_err: F) -> HandleErrorsIter<Self, F>
+    where
+        Self: Iterator<Item = Result<SBP>> + Sized,
+        F: FnMut(&crate::Error) -> bool,
+    {
+        HandleErrorsIter::new(self, on_err)
+    }
```

Comment:
wanted to try and capture the common kinds of error handling we've been doing when going from 

```
Iterator<Item = Result<SBP>> -> Iterator<Item = SBP>
``` 

The bool in `F: FnMut(&crate::Error) -> bool` controls whether to continue after an error. If it's false it stops and you can call `.check_error()` on the iterator see what the error was. Pretty much https://docs.rs/itertools/0.10.0/itertools/fn.process_results.html but not a freestanding function

I think being able to do both is important but not sure this is the best way of doing it.

Reply:
If we're following an existing pattern then I think this is reasonable.  Maybe @john-michaelburke would have an opinion since he's worked on stuff like this in ICBINS.

Reply:
I think this sounds great with the one request of an additional hybrid of "ignore + log errors". Maybe log them as debug instead of warning though or even specifying loglevel?

Reply:
Great idea, I added a level param to `log_errors`. Also replaced the bool with a poor mans https://doc.rust-lang.org/std/ops/enum.ControlFlow.html (it's not stable) to clear up the behavior a bit.